cmd/state/main.go

```diff
@@ -82,7 +82,7 @@ func main() {
 
 	// Set up our output formatter/writer
 	outFlags := parseOutputFlags(os.Args)
-	out, err := initOutput(outFlags, "")
+	out, err := initOutput(outFlags, "", subshell.New(cfg).Shell())
```

Comment:
I'd still like to advocate splitting the shell detection logic out. It's true that `New()` isn't very costly, but it's still doing more than we need it to. Plus we don't know how it will evolve in the future, nor will we be sensitive to this use-case in the future unless we make it explicit.